src/BSeries.jl

```diff
@@ -1556,6 +1556,46 @@ function low_order_energy_preserving(trees, coefficients)
 end
 
 
+"""
+        renormalize_bseries!(series)
+
+This function receives a 'TruncatedBSeries' and returns
+a new 'TruncatedBseries' whose coefficients are divided
+by the symmetry of the corresponding tree.
+"""
+function renormalize_bseries!(series)
+    trees_array = collect(keys(series))
+    l = length(trees_array)
+    for i in 1:l
+        factor = symmetry(trees_array[i])
+        series[trees_array[i]] = series[trees_array[i]]*(1//factor)
+    end
+end
```

Comment:
```suggestion
"""
        renormalize_bseries!(series)

This function modifies a B-series by dividing each coefficient
by the `symmetry` of the corresponding tree.
"""
function renormalize_bseries!(series)
    for t in keys(series)
        series[t] /= symmetry(t)
    end
    return series
end
```

Reply:
@ranocha What would you suggest?  Perhaps after all this function should just return an array of coefficients, to make it clear that they do not represent a B-Series in the same way that the `BSeries` object does?

Reply:
Or we could do away with this function entirely and just divide by the symmetry when needed?  I suppose that is what is done in other parts of the code, like when we print a B-Series?

Reply:
Okey, so we don't want to modify the original BSeries but only to save the renormalized version in another variable and work with this last one, right?

Reply:
I'm not sure, to be honest. This is an ugly problem. As far as I see, we have the following options.

1. Just use this method as it is and tell everyone that they need to be careful.
2. Return an array of normalized coefficients - but we lose the direct association with trees in this case.
3. Add another field to the existing implementation(s) of B-series like `normalized::Bool` defaulting to `true`. Check this field whenever we perform operations on the B-series and throw an error if required. Change this field to `false` when `renormalize!` is called. The function `renormalize!` could switch between both modes.
4. Tell people that they need to renormalize the coefficients themselves. That's essentially what we are doing right now, e.g., in

https://github.com/ranocha/BSeries.jl/blob/04b2f999612dea844e05febea33dfc2dbac6165d/src/BSeries.jl#L361L366

Reply:
How about we go with option 1 for now, but create an issue to continue discussing this?

Reply:
Yes, sounds good to me :+1:

Reply:
https://github.com/ranocha/BSeries.jl/issues/129